### PR TITLE
LVM-310: Implement new/edit student form

### PR DESCRIPTION
**Pull request by vcantu**
[LVM-310](https://cs5500-jira.ccs.neu.edu/browse/LVM-310)
[LVM-306](https://cs5500-jira.ccs.neu.edu/browse/LVM-306)

This PR builds out the new/edit student form

Fixes some student fields to have the exact same name as the fields in tutors for consistency

### CODE REVIEW

File: app/controllers/students_controller.rb [hunk 1]
```diff
@@ -21,6 +22,7 @@ def edit
 
   def create
     @student = Student.new(student_params)
+    calculate_preferences(params)
```
Comment (nabeel-) [inline app/controllers/students_controller.rb]: You'll want to do this before the `Student.new` line. That line will create a new `Student` but does not save it to the database yet so the params are already used and your calculations will not be used. If you're going to modify `params` then you'll also want to use `params` to create the object and not `student_params`.

File: app/controllers/students_controller.rb [hunk 2]
```diff
@@ -31,6 +33,7 @@ def create
 
   def update
     @student = Student.find(params[:id])
+    calculate_preferences(params)
```
Comment (nabeel-) [inline app/controllers/students_controller.rb]: If you're going to use `student_params` to update the student then you need to `calculate_preferences` on them and not `params`, or the other way around works also.

File: db/migrate/20170331012936_update_student_field_names.rb [hunk 3]
```diff
@@ -0,0 +1,15 @@
+class UpdateStudentFieldNames < ActiveRecord::Migration[5.0]
+  def change
+    rename_column :students, :alternate_number, :other_phone
+    rename_column :students, :emergency_name, :emergency_contact_name
+    rename_column :students, :emergency_number, :emergency_contact_phone
+    rename_column :students, :dob, :date_of_birth
+    rename_column :students, :is_hispanic, :hispanic_or_latino
+    rename_column :students, :origin_country, :country_of_birth
+    rename_column :students, :preferred_contact, :preferred_contact_method
+    add_column :students, :referral_other, :referral_other
```
Comment (zack) [inline db/migrate/20170331012936_update_student_field_names.rb]: `:students, :referral_other, :string`?

File: app/controllers/students_controller.rb [hunk 4]
```diff
@@ -123,7 +131,8 @@ def student_params
       :cdbg_legal_resident,
       :cdbg_female_head_of_household,
       :cdbg_household_size,
-      :cdbg_household_income
+      :cdbg_household_income,
+      all_tags: []
```
Comment (zack) [inline app/controllers/students_controller.rb]: You're still missing quite a few values here.

File: app/views/students/_form.html.erb [hunk 5]
```diff
@@ -0,0 +1,393 @@
...
+      </div>
+    </div>
+  </section>
+
+    <hr>
+
+    <div class="row">
+      <div class="col-md-12">
+        <div class="form-group">
+          <label class="control-label"> Contact Method Preferences </label>
```
Comment (zack) [inline app/views/students/_form.html.erb:172]: This is the wrong style

**Comment by zack** [comment]
A bunch of the fields empty out when the form fails. Take a look at the tutor form (which doesn't do this). I suspect it's because the parameters aren't being permitted, but I haven't looked too deep.

**Comment by vcantu** [comment]
@zack Yes working on that now, it's an issue with using the wrong params in the student controller

File: app/views/students/_form.html.erb (hunk 5 above)
Comment (vcantu) [reply app/views/students/_form.html.erb:172]: I think Contact Method Preferences is a sub-section of the Personal Information preferences, hence the different style.  The style is the same in the tutor from

File: app/controllers/students_controller.rb (hunk 4 above)
Comment (vcantu) [reply app/controllers/students_controller.rb]: added them

File: db/migrate/20170328215033_update_new_student_fields.rb [hunk 6]
```diff
@@ -19,7 +19,7 @@ def change
     add_column :students, :cdbg_household_size, :integer
     add_column :students, :cdbg_household_income, :integer
     add_column :students, :intake_date, :date
-
```
Comment (zack) [inline db/migrate/20170328215033_update_new_student_fields.rb:22]: please revert this change.

**Review by zack — APPROVED** [review]
let's merge it!
